mule-extensions-api/src/main/java/org/mule/runtime/extension/api/security/SecurityContextHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.security;
+
+import org.mule.runtime.api.security.Authentication;
+import org.mule.runtime.api.security.SecurityContext;
+import org.mule.runtime.api.security.SecurityException;
+import org.mule.runtime.api.security.SecurityProviderNotFoundException;
+import org.mule.runtime.api.security.UnknownAuthenticationTypeException;
+
+import java.util.List;
+
+/**
+ * This handler allows to configure the current {@link SecurityContext} based on a given {@link Authentication} information.
+ *
+ * @since 1.0
+ */
+public interface SecurityContextHandler {
+
+  /**
+   * Updates the current {@link SecurityContext} with the given {@link Authentication} information.
```

Comment:
explain what a security povider is and give a couple of examples. I have no clue of what they are